### DISCUSSION TITLE
Is there a way to parameterize my R code to pull in the $navbar-bg option from a custom.scss file?

### DISCUSSION THREAD

**Discussion by DamonCharlesRoberts** [quarto-dev/quarto-cli#3263] [Q&A] [answered]
So I have a file called custom.scss where I have set some colors for the theme of a website. The contents are this:

```
/*-- scss:defaults --*/
$navbar-bg: #E14E00;
$navbar-hl: #FBCD98;
$link-color: #762B18;
```

I have a file that has an R code block in it. I would like to use the $navbar-bg color for this code. I know that I could write the yaml for the file to specify a parameters and then access that in my code. I am not sure how to access particular objects in that custom.scss file. So what I currently have is this:

```
---
title: "Blog post"
params:
    primary_color: "custom.scss$navbar-bg"
---
```

But that obviously doesn't work. What would I do to just automatically specify that parameter based on the contents of my custom.scss file?

**Comment by mcanouil** [comment, marked as answer]
I found a way for R using an `rmarkdown` "trick".
Although, I wonder if something more general than the `rmarkdown::metadata` list trick is possible in Quarto, but I think it's a feature that could be very handy.

````md
---
title: Title
format:
  html:
    css: style.scss
---

```{r}
readLines(rmarkdown::metadata[["format"]][["html"]][["css"]])
```
````

<img width="729" alt="image" src="https://user-images.githubusercontent.com/8896044/201204714-9d91ecab-f023-4380-bfd2-6c121fa2f882.png">

---

__*Original post:*__

I drop my idea here, but there is something I quite not understand yet.

````md
---
title: Title
format:
  html:
    css: style.scss
params:
  style-file: "{{< meta format.html.css >}}"
---

Test {{< meta format.html.css >}}

```{r}
params[["style-file"]] == "style.scss"
print(params[["style-file"]])
print("style.scss")
```

````

*EDIT*: Funny, when testing `params[["style-file"]] == "style.scss"`, `params[["style-file"]]` actually contains `{{< meta format.html.css >}}`, only when printed, it becomes `style.scss`.

**Reply by DamonCharlesRoberts** [reply]
Sorry for my slow response... is there a way to access the hexcode for navbar-bg? I'm wondering if I can essentially access that info. The motivation is that I can use that info in the scss file to set the themes for ggplot2 objects.

**Reply by mcanouil** [reply]
With what I suggested, you have access to whatever is inside the scss file, so read and extract what you want.

**Reply by DamonCharlesRoberts** [reply]
This is great, thank you!
